### PR TITLE
Add functionality to retrieve container by its name

### LINKED ISSUE #2302: Make it possible to obtain a reference of type `Container` to services started by Docker Compose
Since `DockerComposeContainer#project` is private, there is currently no way of accessing containers started by Docker Compose.

We should add a public API to access them, something like:
```java
public ContainerState getServiceContainer(String serviceId);
```

We already do something similar here:
https://github.com/testcontainers/testcontainers-java/blob/500fd7404c07712c1f8af2d5f6ba51a048b9f6da/core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java#L256-L263
And store it in:
https://github.com/testcontainers/testcontainers-java/blob/500fd7404c07712c1f8af2d5f6ba51a048b9f6da/core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java#L71

Which means that the implementation should be a simple look up.

This will also solve #1916 and similar requests.

### PR DESCRIPTION
**Pull request by artjomka**
Fixes #2302

### CODE REVIEW

File: core/src/test/java/org/testcontainers/junit/DockerComposeContainerTest.java [hunk 1]
```diff
@@ -33,4 +35,15 @@ public void testGetServicePort() {
...
         assertEquals("Service ports are the same", serviceWithInstancePort, serviceWithoutInstancePort);
     }
+
+    @Test
+    public void testGetContainerByServiceName() {
+        String existingServiceName = "db_1";
+        assertTrue(format("Container should be found by service name %s", existingServiceName),
+            environment.getServiceContainerByName(existingServiceName).isPresent());
+
+        String notExistingServiceName = "db_256";
```
Comment (bsideup) [inline core/src/test/java/org/testcontainers/junit/DockerComposeContainerTest.java]: please create a separate test for the negative scenario

File: core/src/test/java/org/testcontainers/junit/DockerComposeContainerTest.java [hunk 2]
```diff
@@ -33,4 +35,15 @@ public void testGetServicePort() {
         assertNotNull("Port is set for service with instance number", serviceWithoutInstancePort);
         assertEquals("Service ports are the same", serviceWithInstancePort, serviceWithoutInstancePort);
     }
+
+    @Test
+    public void testGetContainerByServiceName() {
+        String existingServiceName = "db_1";
+        assertTrue(format("Container should be found by service name %s", existingServiceName),
```
Comment (bsideup) [inline core/src/test/java/org/testcontainers/junit/DockerComposeContainerTest.java]: could you please also assert some properties of the returned `ContainerState` (e.g. `getExposedPorts()`) to ensure that the returned instance is the correct one (and not Redis for example)

Comment (artjomka) [reply core/src/test/java/org/testcontainers/junit/DockerComposeContainerTest.java]: Added additional check to compare if exported ports belong to correct container we are checking against

File: core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java [hunk 3]
```diff
@@ -479,6 +494,10 @@ public SELF withRemoveImages(RemoveImages removeImages) {
         return self();
     }
 
+    public Optional<ContainerState> getServiceContainerByName(String serviceName) {
```
Comment (bsideup) [inline core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java]: I wonder if we should change the name to something like `getContainerByServiceName`...

@rnorth @kiview WDYT?

Comment (rnorth) [reply core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java]: I think that makes sense, yeah.

Comment (bsideup) [reply core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java]: @artjomka could you please adjust the name of the method? :)

**Comment by bsideup** [comment]
@artjomka merged! 🎉 

Good job - thank you!

**Comment by rnorth** [comment]
Released in 1.13.0! Thanks for the contribution @artjomka  😃

**Pull request merged**